jwst/ramp_fitting/ramp_fit.py

```diff
@@ -474,69 +498,84 @@ def ols_ramp_fit(model, buffsize, save_opt, readnoise_model, gain_model,
+    var_p3, var_both3, slope_int = utils.fix_sat_ramps( sat_0th_group_int,
+                                           var_p3, var_both3, slope_int)
 
-    var_r2 = 1/(s_inv_var_r3.sum(axis=0))
-    var_r2 = utils.rem_nan_inf(var_r2)
-    var_r2[ var_r2 < 0.] = 0.
+    if sat_0th_group_int is not None:
+        del sat_0th_group_int
 
-    err_2d = np.sqrt( var_p2 + var_r2 )
+    # Loop over data integrations to calculate integration-specific pedestal
+    if save_opt:
+        dq_slice = np.zeros((gdq_cube_shape[2],gdq_cube_shape[3]),
+                            dtype=np.uint32)
 
-    # Collect optional results for output
-    opt_res.shrink_crmag(n_int, gdq_cube, imshape, nreads)
+        for num_int in range(0, n_int):
+            dq_slice =  model.get_section('groupdq')[num_int, 0, :, :]
+            opt_res.ped_int[ num_int, :, : ] = \
+                utils.calc_pedestal(num_int, slope_int, opt_res.firstf_int,
+                    dq_slice, nframes, groupgap, dropframes1)
 
-    # Truncate results at the maximum number of segments found
-    opt_res.slope_seg = opt_res.slope_seg[:,:f_max_seg,:,:]
-    opt_res.sigslope_seg = opt_res.sigslope_seg[:,:f_max_seg,:,:]
-    opt_res.yint_seg = opt_res.yint_seg[:,:f_max_seg,:,:]
-    opt_res.sigyint_seg = opt_res.sigyint_seg[:,:f_max_seg,:,:]
-    opt_res.weights = (inv_var_both4[:,:f_max_seg,:,:])**2.
-    opt_res.var_p_seg = var_p4[:,:f_max_seg,:,:]
-    opt_res.var_r_seg = var_r4[:,:f_max_seg,:,:]
+        del dq_slice
 
+    # Collect optional results for output
     if save_opt:
+        gdq_cube = model.groupdq  
+        opt_res.shrink_crmag(n_int, gdq_cube, imshape, nreads)
+        del gdq_cube 
+
+        # Truncate results at the maximum number of segments found
+        opt_res.slope_seg = opt_res.slope_seg[:,:f_max_seg,:,:]
+        opt_res.sigslope_seg = opt_res.sigslope_seg[:,:f_max_seg,:,:]
+        opt_res.yint_seg = opt_res.yint_seg[:,:f_max_seg,:,:]
+        opt_res.sigyint_seg = opt_res.sigyint_seg[:,:f_max_seg,:,:]
+        opt_res.weights = (inv_var_both4[:,:f_max_seg,:,:])**2.
+        opt_res.var_p_seg = var_p4[:,:f_max_seg,:,:]
+        opt_res.var_r_seg = var_r4[:,:f_max_seg,:,:]
+
         opt_model = opt_res.output_optional(model, effintim)
     else:
         opt_model = None
 
-    # Divide slopes by total (summed over all integrations) effective
-    #   integration time to give count rates.
-    c_rates = slope_dataset2 / effintim
+    if inv_var_both4 is not None:
+        del inv_var_both4
 
-    # Compress all integration's dq arrays to create 2D PIXELDDQ array for
-    #   primary output
-    final_pixeldq = dq_compress_final(dq_int, n_int)
+    if var_p4 is not None:
+        del var_p4
+
+    if var_r4 is not None:
+        del var_r4
+
+    if inv_var is not None:
+        del inv_var
+
+    if pixeldq is not None:  
+        del pixeldq
 
     # For multiple-integration datasets, will output integration-specific
```

Comment:
Initialization of 'int_times' is not assured at this point.  In fact, interactive testing using the niriss/test_ramp_fit nightly test triggered an UnboundLocalError Exception at this point due to an uninitialized variable 'int_times'.  Adding the line 'int_times = None' prior to the 'if n_int' block will resolve this problem.